### PR TITLE
eth/fetcher: fix fetcher timeout

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
This PR adds a test and fixes a corresponding bug in the tx fetcher

### CODE REVIEW

**Comment by fjl** [comment]
Damn, sneaky. Please change the constant type to `time.Duration`, it will make it easier to reason about.

**Comment by fjl** [comment]
(sorry wrong button)

**Review by holiman — APPROVED** [review]
I pushed a commit removing some remaining `int64->time.Time` coercion. LGTM now, ltgy?

**Comment by holiman** [comment]
```
ok  	github.com/ethereum/go-ethereum/eth/downloader	113.460s
?   	github.com/ethereum/go-ethereum/eth/ethconfig	[no test files]
--- FAIL: TestTransactionForgotten (0.00s)
    tx_fetcher_test.go:1563: transaction should be evicted by this point
FAIL
FAIL	github.com/ethereum/go-ethereum/eth/fetcher	43.465s

```
Appveyor/windows -- the new test is too much reliant on a non-sucky cpu.

**Comment by MariusVanDerWijden** [comment]
Fixed the test, I think your change slightly changed the semantic from < to <=  which broke the test

**Comment by holiman** [comment]
```
eth/fetcher/tx_fetcher_test.go:2008:3: not enough arguments in call to NewTxFetcher
```
Maybe needs a rebase or something?

File: eth/fetcher/tx_fetcher_test.go [hunk 1]
```diff
@@ -1549,8 +1549,10 @@ func TestTransactionForgotten(t *testing.T) {
 		},
 		func(string, []common.Hash) error { return nil },
 	)
-	go fetcher.loop()
-	// Create a TX which is 5 minutes old, and one which is recent
+	fetcher.Start()
+	defer fetcher.Stop()
+	// Create a TXuno
```
Comment (MariusVanDerWijden) [inline eth/fetcher/tx_fetcher_test.go:1554]: lint

File: eth/fetcher/tx_fetcher_test.go [hunk 2]
```diff
@@ -1535,3 +1535,38 @@ func containsHash(slice []common.Hash, hash common.Hash) bool {
...
+				errs[i] = txpool.ErrUnderpriced
+			}
+			return errs
+		},
+		func(string, []common.Hash) error { return nil },
+	)
+	fetcher.Start()
+	defer fetcher.Stop()
+	// Create a TXuno
+	//which is 5 minutes old, and one which is recent
```
Comment (holiman) [inline eth/fetcher/tx_fetcher_test.go:1555]:
```suggestion
	// Create one TX which is 5 minutes old, and one which is recent
```

**Review by MariusVanDerWijden — COMMENTED** [review]
LGTM

**Review by lightclient — APPROVED** [review]

**Comment by anewt95** [comment]
[]()

**Comment by NorutoSDK** [comment]
>>> w3.eth.get_block('latest')
{'difficulty': 1,
 'gasLimit': 6283185,
 'gasUsed': 0,
 'hash': HexBytes('0x53b983fe73e16f6ed8178f6c0e0b91f23dc9dad4cb30d0831f178291ffeb8750'),
 'logsBloom': HexBytes('0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000'),
 'miner': '0x0000000000000000000000000000000000000000',
 'mixHash': HexBytes('0x0000000000000000000000000000000000000000000000000000000000000000'),
 'nonce': HexBytes('0x0000000000000000'),
 'number': 0,
 'parentHash': HexBytes('0x0000000000000000000000000000000000000000000000000000000000000000'),
 'proofOfAuthorityData': HexBytes('0x0000000000000000000000000000000000000000000000000000000000000000dddc391ab2bf6701c74d0c8698c2e13355b2e4150000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000'),
 'receiptsRoot': HexBytes('0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421'),
 'sha3Uncles': HexBytes('0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347'),
 'size': 622,
 'stateRoot': HexBytes('0x1f5e460eb84dc0606ab74189dbcfe617300549f8f4778c3c9081c119b5b5d1c1'),
 'timestamp': 0,
 'totalDifficulty': 1,
 'transactions': [],
 'transactionsRoot': HexBytes('0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421'),
 'uncles': []}

**Comment by NorutoSDK** [comment]
Overview[](https://web3py.readthedocs.io/en/v6.6.1/overview.html#overview)

The purpose of this page is to give you a sense of everything web3.py can do and to serve as a quick reference guide. You’ll find a summary of each feature with links to learn more. You may also be interested in the [Examples](https://web3py.readthedocs.io/en/v6.6.1/examples.html#examples) page, which demonstrates some of these features in greater detail.

Configuration[](https://web3py.readthedocs.io/en/v6.6.1/overview.html#configuration)

After installing web3.py (via pip install web3), you’ll need to configure a provider endpoint and any middleware you want to use beyond the defaults.

Providers[](https://web3py.readthedocs.io/en/v6.6.1/overview.html#providers)

[Providers](https://web3py.readthedocs.io/en/v6.6.1/providers.html) are how web3.py connects to a blockchain. The library comes with the following built-in providers:

[IPCProvider](https://web3py.readthedocs.io/en/v6.6.1/providers.html#web3.providers.ipc.IPCProvider) for connecting to ipc socket based JSON-RPC servers.
[HTTPProvider](https://web3py.readthedocs.io/en/v6.6.1/providers.html#web3.providers.rpc.HTTPProvider) for connecting to http and https based JSON-RPC servers.
[WebsocketProvider](https://web3py.readthedocs.io/en/v6.6.1/providers.html#web3.providers.websocket.WebsocketProvider) for connecting to ws and wss websocket based JSON-RPC servers.
[AsyncHTTPProvider](https://web3py.readthedocs.io/en/v6.6.1/providers.html#web3.providers.async_rpc.AsyncHTTPProvider) for connecting to http and https based JSON-RPC servers.

**Pull request merged**